File: app/src/main/java/com/fj/temanteman/FriendDetailActivity.java
package com.fj.temanteman;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.fj.temanteman.data.Consts;
import com.fj.temanteman.database.FriendHelper;

public class FriendDetailActivity extends AppCompatActivity implements View.OnClickListener {

    private FriendHelper friendHelper;
    private int id;
    private String phone;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_friend_detail);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);

        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(this);

        ImageView imgDelete = findViewById(R.id.img_delete);
        imgDelete.setOnClickListener(this);

        String initial = getIntent().getStringExtra(Consts.KEY_INITIAL);
        String name = getIntent().getStringExtra(Consts.KEY_NAME);
        String major = getIntent().getStringExtra(Consts.KEY_MAJOR);
        phone = getIntent().getStringExtra(Consts.KEY_PHONE);
        id = getIntent().getIntExtra(Consts.KEY_ID, 0);

        TextView txtName = findViewById(R.id.txt_name);
        TextView txtMajor = findViewById(R.id.txt_major);

        getSupportActionBar().setTitle(initial);
        txtMajor.setText(major);
        txtName.setText(name);

        friendHelper = new FriendHelper(this);
        friendHelper.open();
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.fab:
                Intent intent = new Intent(this, SendMessageActivity.class);
                intent.putExtra(Consts.KEY_PHONE, phone);
                startActivity(intent);
                break;
            case R.id.img_delete:
                friendHelper.delete(id);
                Toast.makeText(this, "Friend deleted", Toast.LENGTH_SHORT).show();
                finish();
                break;
        }
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }
}

File: app/src/main/java/com/fj/temanteman/FriendDetailActivity.kt
package com.fj.temanteman

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

import com.fj.temanteman.data.Const
import com.fj.temanteman.database.FriendHelper
import com.fj.temanteman.models.Friend
import kotlinx.android.synthetic.main.activity_friend_detail.*
import kotlinx.android.synthetic.main.item_friend.*

class FriendDetailActivity : AppCompatActivity(), View.OnClickListener {

    private lateinit var friendHelper: FriendHelper
    private lateinit var friend: Friend

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_friend_detail)
        setSupportActionBar(toolbar)

        friend = intent.getParcelableExtra(Const.KEY_FRIEND)

        supportActionBar?.title = friend.initial
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setDisplayShowHomeEnabled(true)

        fab.setOnClickListener(this)

        img_delete.setOnClickListener(this)

        txt_major.text = friend.major
        txt_name.text = friend.name

        friendHelper = FriendHelper(this)
        friendHelper.open()
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.fab -> {
                val intent = Intent(this, SendMessageActivity::class.java)
                intent.putExtra(Const.KEY_PHONE, friend.phone)
                startActivity(intent)
            }
            R.id.img_delete -> {
                friendHelper.delete(friend.id)
                Toast.makeText(this, "Friend deleted", Toast.LENGTH_SHORT).show()
                finish()
            }
        }
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }
}
